testing/cli/test_sct_run_batch.py

```diff
@@ -61,70 +59,73 @@ def test_config_extra_value_warning():
+def test_non_executable_task(tmp_path, dummy_script):
     data_path = sct_test_path()
-    with \
-        NamedTemporaryFile('w', suffix='.sh') as script,\
-            TemporaryDirectory() as out:
-
-        script_text = """
-        #!/bin/bash
-        echo $SUBJECT
-        """
-        script.write(dedent(script_text)[1:]) #indexing removes beginning newline
+    out = tmp_path / 'out'
+
+    script_text = """
+    #!/bin/bash
+    echo $SUBJECT
+    """
+    with open(tmp_path / "dummy_nonexectuable_script.sh", 'w') as script:
+        script.write(dedent(script_text)[1:])  # indexing removes beginning newline
         script.flush()
 
-        assert not os.access(script.name, os.X_OK), "Script already executable"
+    # The assertion below is meant to ensure that the script is not executable at first,
+    # so that we can test whether `sct_run_batch` properly changes the permissions. However,
+    # checking against os.X_OK isn't compatible with Windows. So, I've commented it out:
+
+    # assert not os.access(script.name, os.X_OK), "Script already executable"
+
+    # More broadly speaking, however: Don't the other tests also accomplish this same functionality?
+    # i.e. Wouldn't _all_ dummy scripts we create start out as non-executable? I suspect that this test
+    # could be combined with the other tests and there wouldn't be a meaningful loss of coverage.
 
-        sct_run_batch.main(['-include', '^t.*',
-                            '-subject-prefix', '',
-                            '-path-data', data_path, '-path-out', out,
-                            '-script', script.name,
-                            '-continue-on-error', 0])
+    sct_run_batch.main(['-include', '^t.*',
+                        '-subject-prefix', '',
+                        '-path-data', data_path, '-path-out', str(out),
+                        '-script', script.name,
+                        '-continue-on-error', 0])
 
 
-def test_no_sessions():
+def test_no_sessions(tmp_path, dummy_script):
     # Test that sessions ('ses') can be separated so that sct_run_batch can process each session folder separately.
-    with TemporaryDirectory() as data,\
-            TemporaryDirectory() as out,\
-            NamedTemporaryFile('w', suffix='.sh') as script:
-        # Create dummy BIDS directory with sessions
-        os.makedirs(os.path.join(data, 'sub-01', 'anat'))
-        os.makedirs(os.path.join(data, 'sub-02', 'anat'))
-        write_dummy_script(script)
-        sct_run_batch.main(['-path-data', data, '-path-out', out, '-script', script.name])
-        file_log = glob.glob(os.path.join(out, 'log', '*sub-01.log'))[0]
-        assert 'sub-01' in open(file_log, "r").read()
-
-
-def test_separate_sessions():
+    data = tmp_path / 'data'
+    data.mkdir()
+    out = tmp_path / 'out'
+    # Create dummy BIDS directory with sessions
+    os.makedirs(os.path.join(data, 'sub-01', 'anat'))
+    os.makedirs(os.path.join(data, 'sub-02', 'anat'))
```

Comment:
Since `tmp_path` is a pathlib-type object, you could use the same idiom as above, `data / 'sub-01' / 'anat'` instead of `os.path.join(...)`. I believe `os.makedirs()` will accept it as-is, even without wrapping it in a `str(...)`.